File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/revenuegoal/RevenueGoal.kt
package com.ongo.domain.revenuegoal

import java.time.LocalDate
import java.time.LocalDateTime

data class RevenueGoal(
    val id: Long = 0,
    val workspaceId: Long,
    val name: String,
    val targetAmount: Long,
    val currentAmount: Long = 0,
    val currency: String = "KRW",
    val period: String,
    val platform: String = "ALL",
    val status: String = "ACTIVE",
    val progress: Int = 0,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val createdAt: LocalDateTime = LocalDateTime.now(),
)

File: frontend/src/types/revenueGoal.ts
export type GoalStatus = 'ACTIVE' | 'COMPLETED' | 'EXPIRED'

export interface RevenueGoal {
  id: number
  name: string
  targetAmount: number
  currentAmount: number
  currency: string
  period: string
  platform: string
  status: GoalStatus
  progress: number
  startDate: string
  endDate: string
}

export interface RevenueGoalMilestone {
  id: number
  goalId: number
  label: string
  targetAmount: number
  reached: boolean
  reachedAt?: string
}

export interface RevenueGoalSummary {
  totalGoals: number
  activeGoals: number
  avgProgress: number
  totalTarget: number
  totalCurrent: number
}
